File: chapters/02_channel_adapter_registry/v1_if_routing.ts
// TypeScript mirror for Chapter 02: Channel Adapter Registry (v1)
// Run with: npx ts-node v1_if_routing.ts

function routeAdapter(channel: string): string {
  const name = channel.toLowerCase();
  if (name === "telegram") return "telegram_adapter";
  if (name === "discord") return "discord_adapter";
  if (name === "webchat") return "webchat_adapter";
  return "default_adapter";
}

function dispatchMessage(channel: string, text: string): string {
  const adapter = routeAdapter(channel);
  return `${adapter}:handled:${text}`;
}

function main(): void {
  console.log(dispatchMessage("telegram", "hello"));
}

main();

File: chapters/02_channel_adapter_registry/v1_if_routing.py
from __future__ import annotations


def route_adapter(channel: str) -> str:
    name = channel.lower()
    if name == "telegram":
        return "telegram_adapter"
    if name == "discord":
        return "discord_adapter"
    if name == "webchat":
        return "webchat_adapter"
    return "default_adapter"


def dispatch_message(channel: str, text: str) -> str:
    adapter = route_adapter(channel)
    return f"{adapter}:handled:{text}"


def main() -> None:
    print(dispatch_message("telegram", "hello"))


if __name__ == "__main__":
    main()
